Stripe/STPPaymentContext.m

```diff
@@ -598,15 +598,14 @@ three_d_secure types of required or optional (not_supported are filtered earlier
                                                                                currency:self.paymentCurrency
                                                                               returnURL:self.configuration.returnURLBlock().absoluteString
                                                                                    card:cardSource.stripeID];
+    threeDSecureParams.metadata = self.sourceInformation.metadata;
 
     STPSourceCompletionBlock threeDSSourceCompletion = ^(STPSource * _Nullable threeDSSource, NSError * _Nullable error) {
         if (error
             || threeDSSource == nil
             || threeDSSource.flow != STPSourceFlowRedirect) {
             if (!threeDSecureRequired
-                && error.domain == StripeDomain
-                && error.code == STPCardError
-                && [error.userInfo[STPCardErrorCodeKey] isEqualToString:STPPaymentMethodNotAvailable]) {
+                && [error.userInfo[STPStripeErrorCodeKey] isEqualToString:@"payment_method_not_available"]) {
```

Comment:
I think all the codes should still be converted to constants. I'm not a huge fan of just checking for the raw string values. Is this not still a card error? (given that 3ds sources are cards). At the least we should define string constants for the errors we check for I think.